File: backend/app/schemas/issue_dependency.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.issue_dependency import DependencyType


class IssueDependencyCreate(BaseModel):
    target_issue_id: int
    dependency_type: DependencyType


class IssueDependencyResponse(BaseModel):
    id: int
    source_issue_id: int
    target_issue_id: int
    dependency_type: DependencyType
    created_at: datetime

    model_config = ConfigDict(
        from_attributes=True,
    )

File: frontend/src/types/issueDependency.ts
export type DependencyType =
  | "BLOCKS"
  | "RELATED"
  | "DUPLICATE";



export interface IssueDependency {

  id: number;

  source_issue_id: number;

  target_issue_id: number;

  dependency_type: DependencyType;

  created_at: string;

}



export interface CreateDependencyRequest {

  target_issue_id: number;

  dependency_type: DependencyType;

}